engine/flink/components/table/src/main/scala/pl/touk/nussknacker/engine/flink/table/utils/DataTypeFactoryHolder.scala

```diff
@@ -0,0 +1,46 @@
+package pl.touk.nussknacker.engine.flink.table.utils
+
+import org.apache.flink.table.api.{EnvironmentSettings, TableConfig}
+import org.apache.flink.table.catalog.{
+  CatalogManager,
+  CatalogStoreHolder,
+  DataTypeFactory,
+  GenericInMemoryCatalog,
+  GenericInMemoryCatalogStore
+}
+
+import scala.util.Using
+
+object DataTypeFactoryHolder {
```

Comment:
`DataTypeFactorySimulatorHolder` + comment that it is not a real `DataTypeFactory` that will be used on Flink but for purpose of simulation of type resolving in various context of usages